lib/project_types/extension/commands/serve.rb

```diff
@@ -7,8 +7,20 @@ class Serve < ExtensionCommand
       NPM_SERVE_COMMAND = %w(run-script server)
 
       def call(_args, _command_name)
+        if argo_admin?
+          ShopifyCli::Tasks::EnsureEnv.call(@ctx, required: [:api_key, :secret, :shop])
+          ShopifyCli::Tasks::EnsureDevStore.call(@ctx)
+        end
+
         CLI::UI::Frame.open(@ctx.message("serve.frame_title")) do
-          success = ShopifyCli::JsSystem.call(@ctx, yarn: YARN_SERVE_COMMAND, npm: NPM_SERVE_COMMAND)
+          yarn_serve_command = YARN_SERVE_COMMAND
+          npm_serve_command = NPM_SERVE_COMMAND
+          if argo_admin?
+            serve_args = %W(--shop=#{project.env.shop})
+            yarn_serve_command += serve_args
+            npm_serve_command = npm_serve_command + %w(--) + serve_args
```

Comment:
Can we do this here?
```suggestion
            npm_serve_command += %w(--) + serve_args
```

Reply:
Fixed!